src/actions/firestore.js

```diff
@@ -205,7 +213,19 @@ export const setListeners = (firebase, dispatch, listeners) => {
       'Listeners must be an Array of listener configs (Strings/Objects)',
     );
   }
-  return listeners.map(listener => setListener(firebase, dispatch, listener));
+
+  listeners.forEach(listener => {
+    const path = getQueryName(listener);
+    const oldListenerCount = pathListenerCounts[path] || 0;
+    pathListenerCounts[path] = oldListenerCount + 1;
```

Comment:
Great work here.
One thought: There is an internal `firebase._` variable where `firebase._.listeners` exists to store this. Also wondering if there could be a situation where there are multiple instances of this variable since we don't make a singleton (i.e. only setting if it is not yet defined).

All of that said, maybe this should live in redux state?